### PR TITLE
mediation policy enhancements

### PR DESCRIPTION
**Pull request by isharac**
moved download logic to shared location 
made download option per policy basis
add delete option
Screen Shot 2019-09-17 at 11 56 45 AM

### CODE REVIEW

**Comment by bhathiya** [comment]
shall we optimize the positioning to reduce white spacing? and there are some alignment issues that should be fixed.

**Comment by isharac** [comment]
styling will be done by @erangatl for this UI.
He will send a separate PR with the changes.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Edit/EditFaultMediationPolicy.jsx [hunk 1]
```diff
@@ -250,13 +212,49 @@ function EditFaultMediationPolicy(props) {
...
+    * Handles the custom mediation policy delete.
+    * @param {any} policyToDelete policy file id that is to be deleted.
+    */
+    function deleteCustomMediationPolicy(policyToDelete) {
+        const promisedGetContent = API.deleteMediationPolicy(policyToDelete, api.id);
+        promisedGetContent
+            .then(() => {
+                setFaultSeqCustom(faultSeqCustom.filter(seq => seq.id !== policyToDelete));
+                Alert.info(<FormattedMessage
+                    id='Apis.Details.MediationPolicies.Edit.EditInMediationPolicy.delete.success'
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Edit/EditFaultMediationPolicy.jsx]: Id needs to be changed as `Apis.Details.MediationPolicies.Edit.EditFaultMediationPolicy.delete.success`

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx [hunk 2]
```diff
@@ -88,47 +89,66 @@ function Overview(props) {
...
+                //         Alert.error('Error occurred while updating API mediation policies');
+                //     });
             })
             .catch((errorResponse) => {
                 console.error(errorResponse);
                 Alert.error('Error occurred while retrieving API to save mediation policies');
             });
     };
     const updateInMediationPolicy = (policies) => {
+        // if (policies.name !== NONE) {
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx]: Shall we remove the commented code

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx [hunk 3]
```diff
@@ -88,47 +89,66 @@ function Overview(props) {
...
+                updateAPI(apiData);
+                // const promisedUpdate = api.update(apiData);
+                // promisedUpdate
+                //     .then(() => {
+                //         Alert.info('Mediation Policies updated successfully.');
+                //     })
+                //     .catch((errorResponse) => {
+                //         console.error(errorResponse);
+                //         Alert.error('Error occurred while updating API mediation policies');
+                //     });
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx]: shall we remove the commented code

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Edit/EditFaultMediationPolicy.jsx (hunk 1 above)
Comment (isharac) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Edit/EditFaultMediationPolicy.jsx]: Fixed with ff15600df9442db39832474f05d6294dad0c0f50
Thanks.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx (hunk 2 above)
Comment (isharac) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Overview.jsx]: fixed.

**Pull request merged**